src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -236,75 +223,113 @@ protected override SynthesizedAttributeData TrySynthesizeIsByRefLikeAttribute()
 
         private void CreateEmbeddedAttributesIfNeeded(DiagnosticBag diagnostics)
         {
-            if (this.NeedsGeneratedIsReadOnlyAttribute)
+            if (this.NeedsGeneratedIsReadOnlyAttribute && _lazyIsReadOnlyAttribute is null)
             {
-                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
+                NeedsEmbeddedAttribute = true;
```

Comment:
>NeedsEmbeddedAttribute = true; [](start = 16, length = 30)

Instead of renaming ```CreateEmbeddedAttributeItselfIfNeeded``` and moving the null check here, would it be simple to move this assignment into ```CreateEmbeddedAttributeItselfIfNeeded``` instead?

Reply:
That's better indeed. Thanks

---
In reply to: [215098114](https://github.com/dotnet/roslyn/pull/29180#discussion_r215098114) [](ancestors = 215098114)